### ISSUE TITLE
[a11y] Update aria role on messages area based on type of messages

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#5715]
We should make better use of the aria alert role for the messages area.
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role

Currently all messages have a hard-coded `role="status"`, provided by layouts. If there is an error or a warning, this role should be changed to `alert` so that screen readers will read the message immediately. 

Recommendations from @indigoxela in https://github.com/backdrop/backdrop-issues/issues/5683

* "info" message - no announcement 
* "status" message - role `status` 
* "warning" message - role `alert`
* "error" message - role `alert`

If we have more than one type of message, I'm guessing we set the role based on the most severe message? If there's an warning or error - we set `alert`, if there's not a warning or error, but there is a status - we set `status`. 

The attribute probably shouldn't be added by layouts, but from system where we can set it based on message type. Or we'll need to find a way to pass that information into layouts. (I'm not sure how easy/hard this will be.)

**Comment by yorkshire-pudding** [comment]
@jenlampton @indigoxela 
I looked at how to add to the overall area, but this is hardcoded in many layout templates, so while could be sorted in core, would be problematic to sort across contrib and custom layout templates (maybe that could be a 2.x enhancement?).

I have added a PR (not very polished - may need slightly better structure and code comments) that adds this to the `div` for each message type. It seems to work ok.  If you had both warnings and errors you would get two alert regions; I don't know how that would work in a screen reader.

What do you think about this approach?

**Comment by yorkshire-pudding** [comment]
This was discussed during the Dev meeting on 2026-07-16:
https://www.youtube.com/live/eXui8qD0ZD4?si=zLw5rRhkCAIVMej1&t=3118

I have updated the PR to remove `role="status"` from the surrounding `messages` container in the core layout templates.
